### PR TITLE
Refactor url generation

### PR DESCRIPTION
**Pull request by puskuruk**
# Description

**discord username: afetharita#0001**

**closes #issue**

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.

## Things to check before creating a PR
- [ ] I have inspected my topic, checked.
- [ ] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [ ] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: components/UI/FilterMenu/FilterReasoningMenu.tsx [hunk 1]
```diff
@@ -44,6 +44,19 @@ export interface FilterReasoningMenuProps {
   onChange: (_option: ReasoningFilterMenuOption) => void;
 }
 
+export const getReasoningFilter = (
+  reasoningFilterMenuOption: ReasoningFilterMenuOption
+) => {
+  reasoningFilterMenuOption.type;
+  if (reasoningFilterMenuOption.type === "channel") {
```
Comment (fdemir) [inline components/UI/FilterMenu/FilterReasoningMenu.tsx:51]: bune gerek yok.

File: components/UI/FilterMenu/FilterReasoningMenu.tsx [hunk 2]
```diff
@@ -44,6 +44,19 @@ export interface FilterReasoningMenuProps {
...
+    return undefined;
+  }
+
+  if (reasoningFilterMenuOption.type === "reason") {
+    return reasoningFilterMenuOption.value;
+  }
+};
```
Comment (fdemir) [inline components/UI/FilterMenu/FilterReasoningMenu.tsx:58]: kapatmdan once return; yazman yeterli. en ustteki kosula girmezse bruaya duser.

File: pages/index.tsx [hunk 3]
```diff
@@ -5,163 +5,109 @@ import Drawer from "@/components/UI/Drawer/Drawer";
...
+      const transformedData = data.results ? dataTransformerLite(data) : [];
+      setMarkerData(transformedData);
+    } catch (e) {
+      setError(true);
+      throw new MaintenanceError(t("common:errors.maintenance").toString());
+    } finally {
+      setIsLoading(false);
     }
-  );
+  }, [sendRequest, t, url.href, url.search]);
```
Comment (fdemir) [inline pages/index.tsx:79]: sendRequeste tam olarak neden ihtiyacimiz var? paramlar build edildikten sonra fetch etmek icin mi?

**Comment by giveerr** [comment]
I'm closing this PR because there is a lack of issue and content information. When you edit the document with the correct information, we will reopen this PR.

https://github.com/acikkaynak/deprem-yardim-backend-go/blob/main/.github/PULL_REQUEST_TEMPLATE.md

File: pages/index.tsx [hunk 4]
```diff
@@ -5,163 +5,109 @@ import Drawer from "@/components/UI/Drawer/Drawer";
...
-    coordinatesAndEventType?.ne_lat,
-    coordinatesAndEventType?.ne_lng,
-    coordinatesAndEventType?.sw_lat,
-    coordinatesAndEventType?.sw_lng,
-    newerThanTimestamp,
-    reasoningFilterMenuOption,
-  ]);
+  const { setDevice } = useMapActions();
 
-  const { error, isLoading, isValidating } = useSWR<DataLite | undefined>(
```
Comment (fdemir) [inline pages/index.tsx:100]: swri neden kaldirdin? swr tek basina datayi guncel tutmak icin birkac ozellik iceriyor. refetchOnWindowFocus gibi. emin miyiz? eger eminsek ve baska bir yerde kullanilmiyorsa dependenciesden de ucurulmali.

File: components/UI/Map/LeafletMap.tsx [hunk 5]
```diff
@@ -189,15 +190,28 @@ function LeafletMap() {
...
-    ? DEFAULT_ZOOM
-    : DEFAULT_ZOOM_MOBILE;
+  useEffect(() => {
+    const zoomFromQuery = Number(zoom);
+    if (zoomFromQuery) {
+      setDefaultZoom(zoomFromQuery);
+      return;
+    }
+
+    const zoomParamFromURL = localCoordinatesURL
```
Comment (fdemir) [inline components/UI/Map/LeafletMap.tsx:200]: URLSearchParams kullanilabilir.

**Review by fdemir — CHANGES_REQUESTED** [review]

File: components/UI/FilterMenu/FilterReasoningMenu.tsx (hunk 1 above)
Comment (puskuruk) [reply components/UI/FilterMenu/FilterReasoningMenu.tsx:51]: Uykulu gözlerle 50. satiri unutmuşum :')

Comment (puskuruk) [reply components/UI/FilterMenu/FilterReasoningMenu.tsx:51]: updated

File: pages/index.tsx (hunk 4 above)
Comment (puskuruk) [reply pages/index.tsx:100]: swr'i `locationsURL` icin atmaya devam edeceğiz çünkü `${BASE_URL}/feeds/${locationId}` gibi bir istegi cacheleyebiliriz fakat areas icin attigimiz request'te kullandigimiz `lat` `lng` degerlerinin precisioni cok yuksek ve ayni istegi iki kere atmamız cok düşük bir ihtimal. Ayrıca areas requestinin her zaman güncel olmasını istiyoruz.

swr'in dokumantasyonundan `refetchOnWindowFocus`'i okuyacağım. Dediğin gibi kullanmamız gereken bir seyse swr'a geri çekebilirim.

File: components/UI/Map/LeafletMap.tsx (hunk 5 above)
Comment (puskuruk) [reply components/UI/Map/LeafletMap.tsx:200]: *kullanilmali +1

Comment (puskuruk) [reply components/UI/Map/LeafletMap.tsx:200]: Fakat o bu pr'in konusu degil.

File: components/UI/FilterMenu/FilterReasoningMenu.tsx (hunk 2 above)
Comment (puskuruk) [reply components/UI/FilterMenu/FilterReasoningMenu.tsx:58]: Bu kod parcasini baska bir dosyadan tasimistim. Okumadan tasimisim. Okuyunca switch/case'in daha uygun oldugunu gordum. Implemente edip kodu pushladim.

File: pages/index.tsx (hunk 3 above)
Comment (puskuruk) [reply pages/index.tsx:79]: Evet, paramlar build olduktan sonra veya istediğimiz baska bir zaman fetch edebilmek icin

Comment (puskuruk) [reply pages/index.tsx:79]: Oncesinde bunu URL'i set ederek yapıyorduk

**Pull request closed without merging**